test/unit/base-plugin-test.js

```diff
@@ -5,16 +5,47 @@ const Rule = require('./../../lib/rules/base');
 const { readdirSync, existsSync, readFileSync } = require('fs');
 const { join, parse: parsePath } = require('path');
 const ruleNames = Object.keys(require('../../lib/rules'));
+const Project = require('../helpers/fake-project');
+const EditorConfigResolver = require('../../lib/get-editor-config');
 
 describe('base plugin', function() {
+  let project, editorConfigResolver;
+  beforeEach(() => {
+    project = Project.defaultSetup();
+
+    editorConfigResolver = new EditorConfigResolver();
+    editorConfigResolver.resolveEditorConfigFiles();
+  });
+
+  afterEach(async () => {
+    await project.dispose();
+  });
+
   function runRules(template, rules) {
     let ast = parse(template);
 
     for (let ruleConfig of rules) {
       let { Rule } = ruleConfig;
-      let options = Object.assign({}, ruleConfig, {
-        moduleName: 'layout.hbs',
-        rawSource: template,
+      let options = Object.assign(
+        {},
+        {
+          filePath: 'layout.hbs',
+          moduleId: 'layout',
+          moduleName: 'layout',
+          rawSource: template,
+          ruleNames,
+        },
+        ruleConfig
+      );
+
+      options.configResolver = Object.assign({}, ruleConfig.configResolver, {
+        editorConfig: () => {
+          if (!options.filePath) {
+            return {};
+          }
+
+          return editorConfigResolver.getEditorConfigData(options.filePath);
+        },
```

Comment:
FYI - Expect future PRs to remove the duplication between this file and `lib/linter.js`. We should have a single canonical mechanism to build a rule instance that ensures the correct arguments are passed.